kythe/cxx/indexer/textproto/analyzer.cc

```diff
@@ -501,6 +589,15 @@ proto::VName TextprotoAnalyzer::CreateAndAddAnchorNode(
   return anchor;
 }
 
+// Adds an anchor node, using the StringPiece's offset relative to
+// `textproto_content_` as the start location.
+proto::VName TextprotoAnalyzer::CreateAndAddAnchorNode(
+    const proto::VName& file_vname, re2::StringPiece sp) {
+  const int begin = sp.begin() - textproto_content_.begin();
```

Comment:
minor: check that the StringPiece resides within textproto_content_ before doing the subtraction:

CHECK(sp.begin() >= textproto_content_ && sb.end() <= textproto_content_.end());

Reply:
:+1: